src/pk/dsa/dsa_import.c

```diff
@@ -116,10 +116,21 @@ int dsa_import(const unsigned char *in, unsigned long inlen, dsa_key *key)
    }
 
 LBL_OK:
-  key->qord = mp_unsigned_bin_size(key->q);
+   key->qord = mp_unsigned_bin_size(key->q);
 
-  if (key->qord >= LTC_MDSA_MAX_GROUP || key->qord <= 15 ||
-      (unsigned long)key->qord >= mp_unsigned_bin_size(key->p) || (mp_unsigned_bin_size(key->p) - key->qord) >= LTC_MDSA_DELTA) {
+   /* quick p, q, g validation, without primality testing */
+   if ((err = dsa_int_validate_pqg(key, &stat)) != CRYPT_OK) {
+      goto LBL_ERR;
+   }
+   if (stat == 0) {
+      err = CRYPT_INVALID_PACKET;
```

Comment:
Looks like we inconsistently return either `CRYPT_INVALID_PACKET` or `CRYPT_INVALID_ARG`, we should always use the same error code. I'd say we should go for `CRYPT_INVALID_PACKET` (an alternative could be to introduce a new error code, but not sure if that's really necessary).